Added range check for parameters ACRO_BAL_PITCH and ACRO_BAL_ROLL in ACRO mode

Lqs66 opened the pull request:
This PR is for fixing issue #30152.

peterbarker commented on ArduCopter/mode_althold.cpp:
```diff
@@ -40,7 +40,7 @@ void ModeAltHold::run()
     // get pilot desired climb rate
     float target_climb_rate = get_pilot_desired_climb_rate();
-    target_climb_rate = constrain_float(target_climb_rate, -get_pilot_speed_dn(), g.pilot_speed_up);
+    target_climb_rate = constrain_float(target_climb_rate, -get_pilot_speed_dn(), constrain_int16(g.pilot_speed_up, 50, 500));
```
Wait, this doesn't look right?

Why are we constraining pilot climb rate between 0.5m/s and 5m/s here?  Some vehicles and scenarios could be *well* outside these ranges!

Lqs66 commented:
> @Lqs66 is this something you want to pursue?
> 
> If so, please remove changes related to pilot speed up/down and change to simply constrain to be >= 0.

I'm sorry for replying to you just now. I determined the parameter range according to the regulations in the ArduPilot documentation.
For the ACRO_BAL_PITCH parameter, the documentation specifies an acceptable input range of 0-3 (https://ardupilot.org/copter/docs/parameters.html#acro-bal-pitch-acro-balance-pitch%EF%BC%89).
For the ACRO_BAL_ROLL parameter, the documentation specifies an acceptable input range of 0-3 (https://ardupilot.org/copter/docs/parameters.html#acro-bal-roll-acro-balance-roll%EF%BC%89).
Is the "change to simply constrain to be >= 0" you mentioned referring to these two parameters?

peterbarker commented:
> I'm sorry for replying to you just now. I determined the parameter range according to the regulations in the ArduPilot documentation. For the ACRO_BAL_PITCH parameter, the documentation specifies an acceptable input range of 0-3 

Yeah, those limits are *absolutely advisory*.  People frequently need to go outside of the recommended ranges.

> Is the "change to simply constrain to be >= 0" you mentioned referring to these two parameters?

Yes - just limit so that you can't specify a negative value, which should have been sufficient to avoid the crash (assuming you're capable of uprighting the vehicle yourself :-)

Lqs66 commented:
> > I'm sorry for replying to you just now. I determined the parameter range according to the regulations in the ArduPilot documentation. For the ACRO_BAL_PITCH parameter, the documentation specifies an acceptable input range of 0-3
> 
> Yeah, those limits are _absolutely advisory_. People frequently need to go outside of the recommended ranges.
> 
> > Is the "change to simply constrain to be >= 0" you mentioned referring to these two parameters?
> 
> Yes - just limit so that you can't specify a negative value, which should have been sufficient to avoid the crash (assuming you're capable of uprighting the vehicle yourself :-)

I have addressed the comments and modified the PR :-)
